### ISSUE TITLE
Can't replicate results

### ISSUE THREAD

**Issue by jbmaxwell** [nnormandin/Conditional_VAE#1]
Hello. 
I can't seem to reproduce your results. I'm on Ubuntu 16.04, python 3.6.2, with tensorflow-gpu backend, on cuda 8. I get very noisy images (number of epochs doesn't seem to make much difference). 
![img8 967](https://user-images.githubusercontent.com/15166432/31050293-ea3732c2-a5fa-11e7-9e40-4a2fec68cc36.jpg)
I can run the jupyter notebook, so it seems to basically run, it's just not learning as it should...

**Comment by nnormandin** [comment]
@jbmaxwell what type of values are you seeing for the different loss metrics?

i was able to train this network pretty rapidly on a dual core laptop CPU. i will investigate today

**Comment by jbmaxwell** [comment]
After 20 epochs: 
```
Epoch 20/20
60000/60000 [==============================] - 7s - loss: 431.0483 - KL_loss: 2.7520e-07 - recon_loss: 431.0483 - val_loss: 444.2565 - val_KL_loss: 1.4404e-05 - val_recon_loss: 444.2565
```

**Comment by nnormandin** [comment]
@jbmaxwell the issue looks like it was caused by changes in the default behavior of some operations in K.backend. The `KL_loss` term should never really go below 4, otherwise the model won't be fitting properly. Please pull the most recent updates and you should be able to get workable results by the second or third epoch.

**Comment by jbmaxwell** [comment]
I couldn't actually pull for some reason (`fatal: Not a git repository`), so I deleted and re-cloned—but I'm getting the same output as before. What are the full specs you're using? Are you also running cuda 8? 
[EDIT: Oh, wait... The diff only shows changes to the python notebook—I've been running `python conditional_vae.py`... I'll trying loading up the notebook.]
[UPDATE: Okay, all good in the notebook. Thanks.]

**Comment by hitzhs** [comment]
It seems beacuse the reconstruction_loss loss function in conditional_vae.py,the y_true and y_pred are in wrong positon, exchange their position, the result will be good
